code/iaas/logic/src/main/java/io/cattle/platform/process/containerevent/ContainerEventCreate.java

```diff
@@ -161,8 +177,38 @@ void scheduleInstance(ContainerEvent event, Instance instance, Map<String, Objec
         objectProcessManager.scheduleProcessInstance(PROCESS_CREATE, instance, makeData());
     }
 
-    Map<String, Object> getInspect(ContainerEvent event) {
-        return CollectionUtils.toMap(DataUtils.getFields(event).get(FIELD_DOCKER_INSPECT));
+    private Map<String, Object> getInspect(ContainerEvent event, Map<String, Object> data) {
+        Object inspectObj = DataUtils.getFields(event).get(FIELD_DOCKER_INSPECT);
+        if (inspectObj == null) {
+            EventVO<Object> inspectEvent = newInspectEvent(null, event.getExternalId());
+            inspectObj = callAgentForInspect(data, inspectEvent);
+        }
+        return CollectionUtils.toMap(inspectObj);
+    }
+
+   private Object callAgentForInspect(Map<String, Object> data, EventVO<Object> inspectEvent) {
```

Comment:
we should probably use type of Event interface instead of EventVO implementation. Shouldn't cause the problem switching because agent.callSync expects the interface.